src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,203 @@
+	const [ errorMessage, setError ] = useState( '' );
+
+	const prevIDs = usePrevious( attributes.restaurantIDs );
+
+	const getLocaleCode = () => {
+		if ( typeof coblocksBlockData.localeCode !== 'undefined' ) {
+			return coblocksBlockData.localeCode;
+		}
+		return '';
+	};
+
+	//searches the opentable reservation network for restaurants with the given name or ID
+	const searchRestaurants = useCallback( debounce( ( token ) => {
+		setIsLoading( true );
+		setError( '' );
+		fetch(
+			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=15' +
+				'&query=' +
+				encodeURIComponent( token )
+		)
+			.then( ( response ) => response.json() )
+			.catch( ( error ) => {
+				setIsLoading( false );
+				setError( __( 'Error connecting to the OpenTable API. Please try again later.', 'coblocks' ) );
+				throw new Error( 'Error connecting to the OpenTable API: ' + error );
+			} )
+			.then( ( json ) => {
+				const results = [];
+				for ( const item in json.items ) {
+					const itemProps = json.items[ item ];
+					const name = `${ decodeURIComponent( itemProps.name ) },  ${ decodeURIComponent( itemProps.addressResponse.city ) }, ${ decodeURIComponent( itemProps.addressResponse.country ) } (${ itemProps.rid })`;
+					results.push( name );
+				}
+				setQueryResults( results );
+				setNoResultsFound( results?.length === 0 );
+				setIsLoading( false );
+			} );
+	}, 400 ), [] );
+
+	useEffect( () => {
+		if ( prevIDs !== attributes.restaurantIDs ) {
+			const restaurantNames = attributes.restaurantIDs.map( ( restaurantObject ) => restaurantObject.name );
+			setRidField( restaurantNames );
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( attributes.language === '' ) {
+			switch ( getLocaleCode().substring( 0, coblocksBlockData.localeCode.indexOf( '_' ) ) ) {
+				case 'fr':
+					props.setAttributes( { language: 'fr-CA' } );
+					break;
+				case 'de':
+					props.setAttributes( { language: 'de-DE' } );
+					break;
+				case 'es':
+					props.setAttributes( { language: 'es-MX' } );
+					break;
+				case 'ja':
+					props.setAttributes( { language: 'ja-JP' } );
+					break;
+				case 'nl':
+					props.setAttributes( { language: 'nl-NL' } );
+					break;
+				case 'it':
+					props.setAttributes( { language: 'it-IT' } );
+					break;
+				default:
+					props.setAttributes( { language: 'en-US' } );
+					break;
+			}
+		}
+	}, [] );
+
+	return (
+		<>
+			<Controls
+				isEditing={ isEditing }
+				setIsEditing={ setIsEditing }
+				{ ...props } />
+			<InspectorControls
+				className={ className }
+				attributes={ attributes }
+				setAttributes={ props.setAttributes }
+			/>
+			<div className={ className }>
+
+				{ ( isEditing ) ? (
+					<Placeholder
+						icon={ <Icon icon={ icon } /> }
+						label={ __( 'OpenTable', 'coblocks' ) }
+						instructions={ __(
+							'Select your OpenTable restaurant(s) to show the reservations widget.',
+							'coblocks'
+						) }
+						isColumnLayout={ true }
+					>
```

Comment:
The `Placeholder` component has a `notices` prop you can pass the notices to and it will render them at the top with proper styling applied to maintain a consistent UX.

https://github.com/WordPress/gutenberg/blob/wp/5.8/packages/components/src/placeholder/index.js#L16-L30

Example from the `core/file` block:

![image](https://user-images.githubusercontent.com/375788/128406563-faea85cb-04a7-4a58-8cce-eec549509fcc.png)

Reply:
resolved in 9d8be93

Reply:
<img width="684" alt="image" src="https://user-images.githubusercontent.com/9936886/128408196-5a5059c5-4284-4e7e-ae7c-e0306b53a91b.png">
It seems to have some odd padding behavior. I can confirm this is not being applied from any of the opentable stylesheets.

Reply:
Both of these styles are coming from core (the extra padding and `display: flex` causing the block not to fully expand to the placeholder size) so going to ignore for now.

Reply:
I think you'll need to utilize the `withNotices` HoC which passes a method to create a notice as well as the UI the prop expect.

File block example:
- `withNotices`
https://github.com/WordPress/gutenberg/blob/wp/5.8/packages/block-library/src/file/edit.js#L313

- Props passed
https://github.com/WordPress/gutenberg/blob/wp/5.8/packages/block-library/src/file/edit.js#L66-L67

- Creating the notice
https://github.com/WordPress/gutenberg/blob/wp/5.8/packages/block-library/src/file/edit.js#L104

- Passing `noticesUI` to `Placeholder`
prop https://github.com/WordPress/gutenberg/blob/wp/5.8/packages/block-library/src/file/edit.js#L203

Reply:
And for reference this is the `withNotices` HoC source: https://github.com/WordPress/gutenberg/blob/wp/5.8/packages/components/src/higher-order/with-notices/index.js

It looks like it's wrapping Notice components within a `NoticeList` component which might include the additional styles.

Reply:
https://user-images.githubusercontent.com/9936886/128420681-f2c1abd3-851a-4fbd-b418-307709db1c22.mov

This worked. The off-center X is driving me crazy though, haha.

Reply:
fixed:
- dismiss button alignment 
- use withNotices HOC instead for UX
in e9bd4e5